gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/java/io/gravitee/gateway/standalone/node/GatewayNodeMetadataResolver.java

```diff
@@ -65,35 +67,81 @@ public class GatewayNodeMetadataResolver implements NodeMetadataResolver {
 
     public Map<String, Object> resolve() {
         final HashMap<String, Object> metadata = new HashMap<>();
-
-        final String installationId = getInstallationId();
-        final Set<String> organizationIds = getOrganizationIds();
-        final Set<String> environmentIds = getEnvironmentIds(organizationIds);
+        metadata.put(META_INSTALLATION, getInstallationId());
+        Set<String> organizationIds;
+        Set<String> environmentIds;
+        if (configuration.useLegacyEnvironmentHrids()) {
+            organizationIds = resolveOrganizationsFromHrids();
+            environmentIds = resolveEnvironmentFromHrids(organizationIds);
+        } else {
+            Set<Environment> environments = resolveEnvironments();
+            organizationIds = new HashSet<>();
+            environmentIds = new HashSet<>();
+            environments.forEach(environment -> {
+                organizationIds.add(environment.getOrganizationId());
+                environmentIds.add(environment.getId());
+            });
+        }
+        metadata.put(META_ORGANIZATIONS, Collections.unmodifiableSet(organizationIds));
+        metadata.put(META_ENVIRONMENTS, Collections.unmodifiableSet(environmentIds));
 
         configuration.tenant().ifPresent(tenant -> metadata.put("tenant", tenant));
-
         configuration.shardingTags().ifPresent(shardingTags -> metadata.put("tags", shardingTags));
-
         configuration.zone().ifPresent(zone -> metadata.put("zone", zone));
 
-        metadata.put(META_INSTALLATION, installationId);
-        metadata.put(META_ORGANIZATIONS, organizationIds);
-        metadata.put(META_ENVIRONMENTS, environmentIds);
-
         return metadata;
     }
 
+    private Set<Environment> resolveEnvironments() {
+        final Optional<List<String>> optEnvironmentsList = configuration.environments();
+        return optEnvironmentsList
+            .stream()
+            .flatMap(Collection::stream)
+            .filter(id -> {
+                if (!validateEnvironmentId(id)) {
+                    log.warn("Environment id [%s] should be an valid UUID ; HRIDs are not supported anymore, ignore it.");
+                    return false;
+                }
+                return true;
+            })
+            .map(id -> {
+                try {
+                    return environmentRepository.findById(id);
+                } catch (TechnicalException e) {
+                    log.warn("Unable to load environment from id {}, ignore it.", id, e);
+                    return Optional.<Environment>empty();
+                }
+            })
+            .filter(Optional::isPresent)
+            .map(Optional::get)
```

Comment:
Can be simplified

```suggestion
            .flatMap(id -> {
                try {
                    return environmentRepository.findById(id).stream();
                } catch (TechnicalException e) {
                    log.warn("Unable to load environment from id {}, ignore it.", id, e);
                    return Stream.<Environment>empty();
                }
            })
```

Reply:
Honestly I don't feel it is more simple.

Reply:
Code as been removed anyway. 👍